mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/permissions/PermissionsManager.java

```diff
@@ -0,0 +1,110 @@
+package com.mapbox.services.android.telemetry.permissions;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+
+import java.util.ArrayList;
+
+/**
+ * Helps request permissions at runtime
+ */
+
+public class PermissionsManager {
+
+  public static final String COARSE_LOCATION_PERMISSION = Manifest.permission.ACCESS_COARSE_LOCATION;
+  public static final String FINE_LOCATION_PERMISSION = Manifest.permission.ACCESS_FINE_LOCATION;
+
+  private final int REQUEST_PERMISSIONS_CODE = 0;
+
+  private Activity activity;
```

Comment:
would it be possible to remove activity completely? it's used for 2 items: permissions granted checks and requesting permissions. Imo I think callers of those method won't mind passing an reference to the Activity.